include/TreeNode.hpp

```diff
@@ -22,40 +23,37 @@ class TreeNode {
     std::vector<TreeNode*>          children{};
     size_t                          clusterSize = 1U;
     std::unordered_set<std::size_t> boundaryVertices{};
-    std::vector<std::size_t> checkVertices{};
+    std::vector<std::size_t>        checkVertices{};
     bool                            marked  = false;
     bool                            deleted = false;
 
-    TreeNode():
-        TreeNode(-1) {}
+    TreeNode() : TreeNode(-1) {}
 
-    explicit TreeNode(const std::size_t& vertexIdx):
-        vertexIdx(vertexIdx) {
+    explicit TreeNode(const std::size_t& vertexIdx) : vertexIdx(vertexIdx) {
         boundaryVertices.emplace(vertexIdx);
     }
 
     /*
- * find using path compression
- */
-    static TreeNode* Find(TreeNode* node) {
-        //std::cout << "in find" << std::endl;
-        auto parent = node->parent;
+     * find using path compression
+     */
+    static TreeNode* Find(TreeNode* node) { // NOLINT(readability-identifier-naming)
+        auto* parent = node->parent;
         while (parent != nullptr && parent->parent != nullptr) {
-            node = parent;
+            node   = parent;
             parent = parent->parent;
         }
         if (parent == nullptr) {
             return node;
-        } else {
+        } else { // NOLINT(readability-else-after-return)
             return parent;
         }
     }
     /*
      * Merge two trees with given roots
      */
-    static void Union(TreeNode* tree1, TreeNode* tree2) {
-        auto        root1 = Find(tree1);
-        auto        root2 = Find(tree2);
+    static void Union(TreeNode* tree1, TreeNode* tree2) { // NOLINT(readability-identifier-naming)
```

Comment:
why not rename it to `union`?

Reply:
its a reserved keyword.  thats why i'd like to keep Find and Union for consistency.

Reply:
Makes sense, didn't think about that 👍🏻